src/Common/tests/Cryptography/AlgorithmImplementations/RSA/RSAFactory.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Security.Cryptography.Rsa.Tests
+{
+    public interface IRSAProvider
+    {
+        RSA Create();
+        RSA Create(int keySize);
```

Comment:
The RSA tests are shared source that gets pulled into each of the three implementation libraries to validate the algorithm functionality.  The only way that I could think of to use something like theory required building some after-everything-else library which could see all of RSACryptoServiceProvider, RSACng, and RSAOpenSsl... and we don't have anything like that currently.

A theory-based approach is how the 4.6 RSA API was validated, but RSACryptoServiceProvider and RSACng were both available to that particular test.  It was nice, and I wished that I could have come up with a way to do it here.

Another approach I considered involved a theory with an abstract driver, but I was concerned that that approach would lead to silent not-run situations (if the class was never extended then it just wouldn't run, and therefore wouldn't fail).  This approach at least turns into compile failures.

If we are okay with the notion of having a test library that lives downstream of multiple parallel libraries to vie them off of each other, I'm happy to do that; but I anticipate that having some "where should we put it?" type discussions, and this approach at least gets test coverage restored in the short term.